minimint-api/src/db/sled_impl.rs

```diff
@@ -1,47 +1,44 @@
 use super::batch::{BatchItem, DbBatch};
-use super::{Database, DatabaseError, DecodingError};
+use super::{Database, DecodingError};
 use crate::db::PrefixIter;
+use anyhow::Result;
 use sled::transaction::TransactionError;
 use tracing::{error, trace};
 
 // TODO: maybe make the concrete impl its own crate
 impl Database for sled::Tree {
-    fn raw_insert_entry(
-        &self,
-        key: &[u8],
-        value: Vec<u8>,
-    ) -> Result<Option<Vec<u8>>, DatabaseError> {
+    fn raw_insert_entry(&self, key: &[u8], value: Vec<u8>) -> Result<Option<Vec<u8>>> {
         Ok(self
             .insert(key, value)
-            .map_err(|e| DatabaseError::DbError(Box::new(e)))?
+            .map_err(anyhow::Error::from)?
```

Comment:
I'm pretty sure this could be just a `?` instead of mapping the error manually.

Reply:
will be fixed in next push, waiting for the discussion on the BitcoinRpc and DbError to resolve tough